### PR TITLE
Port android related articles

### PR DESCRIPTION
**Pull request by adegeo**
In progress....

- Add the following articles:
  - device-manager
  - device-properties
  - hardware-acceleration
  - index
  - device setup

Mac-related content has been commented out.

[internal preview link](https://review.docs.microsoft.com/en-us/dotnet/maui/android/emulator/?branch=pr-en-us-217)

### CODE REVIEW

File: docs/android/emulator/device-manager.md [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+
+When you install the **Mobile development with .NET** workload in Visual Studio, and select the optional **.NET MAUI (Preview)** workload, everything is installed for you. For more information on setting up .NET MAUI with Visual Studio, see [Installation](../../get-started/installation.md).
+
+## Open the device manager
+
+Open the Android Device Manager in Visual Studio from the **Tools** menu by pressing **Tools > Android > Android Device Manager**:
+
+:::image type="content" source="media/device-manager/win/03-tools-menu.png" alt-text="Launching the Device manager from the Tools menu.":::
+
+<!-- TODO: Is this still applicable? -->
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md]: No. The min SDK version is 30: Beginning in August 2021, the Google Play Console requires that new apps target API level 30 (Android 11.0) or higher. Existing apps are required to target API level 30 or higher beginning in November 2021.

So no one should need to install API26.

File: docs/android/emulator/device-manager.md [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+
+01. Open one of the log files and search for an entry describing **Config file path**. For example:
+
+    :::image type="content" source="media/device-manager/win/31-config-file-path.png" alt-text="Config file path in log file.":::
+
+01. Navigate to this folder and open the _user.config_ in a text editor.
+
+01. In _user.config_, find the `<UserSettings>` element and add an **AndroidSdkPath** attribute to it. Set this attribute to the path where the Android SDK is installed on your computer and save the file. For example, `<UserSettings>` would look like the following if the Android SDK was installed at _C:\\Programs\\Android\\SDK_:
+
+    <!-- TODO: Does visual studio handle this some how now? -->
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md]: I'd be inclined to comment out/remove this section, unless you are sure it works. Using a non-default location is going to open up a world of pain.

File: docs/android/emulator/device-manager.md [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+
+01. Restart the AVD.
+
+After these changes are made, the AVD will restart in a state that allows Wi-Fi to work again.
+
+::: zone-end
+
+-->
+
+## Related links
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md]: I'd delete any Related links sections, and instead just ensure the links are featured somewhere in the content. Related links is a hangover from when Xamarin docs weren't on docs.microsoft.com.

File: docs/android/emulator/hardware-acceleration.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+ms.date: 12/02/2021
+---
+
+# Hardware acceleration for better emulator performance (Hyper-V & HAXM)
+
+This article explains how to use your computer's hardware acceleration features to maximize Android emulator performance.
+
+With Visual Studio, you can easily test and debug your .NET MAUI app for Android in situations where an Android device isn't available. However, if hardware acceleration isn't available or enabled, the emulator will run too slow. You can drastically improve the performance of the emulator by enabling hardware acceleration and using special x86 virtual device images.
+
+<!-- TODO: What are the virtualization extensions on the new M1 ARM processors for Mac?
```
Comment (davidbritch) [inline docs/android/emulator/hardware-acceleration.md:16]: No idea!

File: docs/android/emulator/hardware-acceleration.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+           Also, isn't Hypervisor.Framework a macOS-only tech? Why is this at the start?
```
Comment (davidbritch) [inline docs/android/emulator/hardware-acceleration.md]: Yes. It'll be at the start because the original doc will date back to when Mac support was more important than Windows support (it used to be the case that Mac support ruled, while Windows support lagged behind).

File: docs/android/emulator/troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,475 @@
...
+    emulator-5554   device
+    ```
+
+01. If the emulator doesn't appear in this list, start the **Android SDK Manager**, apply all updates, then try launching the emulator again.
+
+### MMIO access error
+
+If the message **An MMIO access error has occurred** is displayed, restart the emulator.
+
+<a name="gps-win"></a>
```
Comment (davidbritch) [inline docs/android/emulator/troubleshooting.md]: Remove stray HTML.

File: docs/android/emulator/troubleshooting.md [hunk 7]
```diff
@@ -0,0 +1,475 @@
...
+
+## Missing Google Play Services
+
+If the emulated Android device doesn't have Google Play Services or Google Play Store installed, you probably created a virtual device that excluded these packages. When you [create a virtual device](device-manager.md), be sure to select one or both of the following options:
+
+- **Google APIs** &ndash; includes Google Play Services in the virtual device.
```
Comment (davidbritch) [inline docs/android/emulator/troubleshooting.md]: Any `&ndash;` shows that the original content dates back to the days when the Xamarin publishing system couldn't handle the dash character!

Comment (adegeo) [reply docs/android/emulator/troubleshooting.md]: I've seen this used in non Xamarin content too.

```
-  < dash
–  < en dash
—  < em dash
```

Don't know if it matters though. ..... OK I just looked it up in the style guide and I think in this case it should be an em dash `&mdash;` on these types of of thing—extra info.

I never really use them though, I can dump these if you want, or change them to em dash. I think it does look better though because it is a longer dash.

**Comment by adegeo** [comment]
@davidbritch I added the debug article and a basic TOC for this set of articles. Just an idea on the TOC.

I'm thinking I'll move this file up a directory into a **deployment** or **debugging** folder.

File: docs/android/emulator/troubleshooting.md (hunk 7 above)
Comment (davidbritch) [reply docs/android/emulator/troubleshooting.md]: My point, which I never explained(!) is that we don't need to specify them using escape codes. Anytime I encounter any dash escape codes I replace them with the an actual em dash character.

**Comment by davidbritch** [comment]
> @davidbritch I added the debug article and a basic TOC for this set of articles. Just an idea on the TOC.
> 
> I'm thinking I'll move this file up a directory into a **deployment** or **debugging** folder.

This actually got discussed in a meeting yesterday, and what the product team is angling for is this structure:

```
Get started
    Setup
         Android
               Emulator
                      ...
               Device
         iOS
              Hot restart
              Remote simulator
              Device
         Windows
```

File: docs/android/emulator/device-manager.md [hunk 8]
```diff
@@ -0,0 +1,424 @@
...
+
+- **Start with Factory Defaults** &ndash; TODO
+
+- **Start with Kernel Logs** &ndash; TODO
+
+- **Download System Image** &ndash; Downloads the Android OS system image for the device, if it's not already downloaded.
+
+- **Reveal in Explorer** &ndash; Opens Windows Explorer and navigates to the folder that holds the files for the virtual device.
+
+- **Repair** &ndash; TODO
```
Comment (adegeo) [inline docs/android/emulator/device-manager.md]: @davidbritch can the product group clarify what these TODO options to?

**Comment by adegeo** [comment]
@davidbritch besides the open discussion on what those other menu options mean, I think this is good to go. I did ping the engineer today about them.

**Review by davidbritch — CHANGES_REQUESTED** [review]
Is it "If the Hyper-V results indicate..." or "If the Hyper-V result indicates...."?

File: docs/TOC.yml [hunk 9]
```diff
@@ -13,6 +13,28 @@
           href: get-started/first-app.md
         - name: Migrate from Xamarin.Forms
           href: get-started/migrate.md
+        - name: Android
+          items:
+            - name: Emulator
+              expanded: true
```
Comment (davidbritch) [inline docs/TOC.yml]: We can drop all the expanded: true now.

File: docs/TOC.yml [hunk 10]
```diff
@@ -13,6 +13,28 @@
...
+                  href: android/emulator/index.md
+                - name: Manage and create virtual devices
+                  href: android/emulator/device-manager.md
+                - name: Edit virtual devices
+                  href: android/emulator/device-properties.md
+                - name: Debug on a virtual device
+                  href: android/emulator/debug-on-emulator.md
+                - name: Troubleshoot common problems
+                  href: android/emulator/troubleshooting.md
+                - name: How to enable hardware acceleration
```
Comment (davidbritch) [inline docs/TOC.yml]: I think this item should go after the overview. It's the most important doc really.

File: docs/android/device/setup.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+
+You'll receive a prompt to trust the computer on your device if you haven't used it for debugging before. You can also check **Always allow from this computer** to prevent requiring this prompt each time you connect the device.
+
+:::image type="content" source="media/setup/trust-computer-for-usb-debugging.png" alt-text="Android trust prompt from computer to use USB debugging.":::
+
+If your computer isn't recognizing the device when it's plugged in, try installing a driver for the device. Consult your device manufacturer's support documentation. You can also try installing the Google USB Driver through the Android SDK Manager:
+
+:::image type="content" source="media/setup/google-usb-driver.png" alt-text="Android SKD Manager with the Google USB device driver selected.":::
+
+## Alternatively enable WiFi debugging
```
Comment (davidbritch) [inline docs/android/device/setup.md]: Maybe drop "Alternatively" from the heading?

File: docs/android/device/setup.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+    adb usb
+    ```
+
+    To see the devices connected to the computer, use the `adb devices` command:
+
+    ```command
+    adb devices
+    ```
+
+## Summary
```
Comment (davidbritch) [inline docs/android/device/setup.md]: Always delete the Summary section.

File: docs/android/device/setup.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+
+This article discussed how to configure an Android device for development by enabling debugging on the device. It also covered how to connect the device to a computer using either USB or WiFi.
+
+## Related Links
```
Comment (davidbritch) [inline docs/android/device/setup.md]: Always delete the related links section.

File: docs/android/emulator/debug-on-emulator.md [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+## Launching the Emulator
+
+Near the top of Visual Studio, there's the **Solution Configurations** drop-down menu that can be used to select **Debug** or **Release** mode. Choosing **Debug** causes the debugger to attach to the application process running inside the emulator after the app starts. Choosing **Release** mode disables the debugger. When in release mode, you'll need to rely on app logging for debugging.
+
+After you've chosen a virtual device from the **Debug Target** device drop-down menu, select either **Debug** or **Release** mode, then select the Play button to run the application:
+
+:::image type="content" source="media/debug-on-emulator/win/vs-config-selection.png" alt-text="Debug and release modes in Visual Studio along with the Play button.":::
```
Comment (davidbritch) [inline docs/android/emulator/debug-on-emulator.md:19]: This is a dark mode screenshot. IDE screenshots are meant to be the default (light) theme, right ("For IDE images, use default colors or themes")?

File: docs/android/device/setup.md (hunk 13 above)
Comment (davidbritch) [reply docs/android/device/setup.md]: If these links are worth keeping, do something like the debug-on-emulator doc that ends with "For more info about....see the following..."?

File: docs/android/emulator/hardware-acceleration.md [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+description: "Learn how to enable hardware acceleration features to maximize Android emulator performance for a .NET MAUI app."
+ms.date: 01/19/2022
+ms.topic: how-to
+---
+
+# How to enable hardware acceleration with Android emulators (Hyper-V & HAXM)
+
+This article explains how to use your computer's hardware acceleration features to maximize Android emulator performance.
+
+With Visual Studio, you can easily test and debug your .NET MAUI app for Android in situations where an Android device isn't available. However, if hardware acceleration isn't available or enabled, the emulator will run too slow. You can drastically improve the performance of the emulator by enabling hardware acceleration and using special x86 virtual device images.
```
Comment (davidbritch) [inline docs/android/emulator/hardware-acceleration.md]: Should all mentions of the device images in this article be saying they should be x86_64?

File: docs/android/emulator/index.md [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+title: "Android Emulator Setup"
+description: "The Android Emulator can be run in a variety of configurations to simulate different devices. This article explains how to prepare the Android Emulator for testing your .NET MAUI app."
+ms.date: 12/02/2021
+---
+
+# Android emulator setup
+
+The Android Emulator can be run in a variety of configurations to simulate different devices. Each configuration is called a _virtual device_. When you deploy and test your app on the emulator, you select a pre-configured or custom virtual device that simulates a physical Android device such as a Nexus or Pixel phone.
+
+The following articles are available to help manage and setup your Android emulator:
```
Comment (davidbritch) [inline docs/android/emulator/index.md]: The debugging article is missing from the list, but seems to be mentioned in the final sentence in the doc.

File: docs/android/emulator/device-manager.md [hunk 17]
```diff
@@ -0,0 +1,426 @@
...
+ms.date: 11/18/2021
+no-loc: ["user.config"]
+---
+
+# Managing virtual devices with the Android Device Manager
+
+This article explains how to use the Android Device Manager to create and configure Android Virtual Devices (AVDs) that emulate physical Android devices. You can use these virtual devices to run and test your app without having to rely on a physical device.
+
+> [!IMPORTANT]
+> Enable hardware acceleration for the android devices. For more information, see [Hardware Acceleration for Emulator Performance](hardware-acceleration.md).
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md]: Capital a on Android.

File: docs/android/emulator/device-manager.md [hunk 18]
```diff
@@ -0,0 +1,426 @@
...
+
+You use the Android Device Manager to create and configure an Android Virtual Devices (AVD) that run in the [Android Emulator](debug-on-emulator.md). Each AVD is an emulator configuration that simulates a physical Android device. This makes it possible to run and test your app in a variety of configurations that simulate different physical Android devices.
+
+:::image type="content" source="media/device-manager/win/01-devices-dialog-sml.png" alt-text="Screenshot of the Android Device Manager in the Devices tab." lightbox="media/device-manager/win/01-devices-dialog.png":::
+
+## Requirements
+
+To use the Android Device Manager, you'll need the following items:
+
+- Visual Studio 2022 version 17.1 Preview 3. Visual Studio Community, Professional, and Enterprise editions are supported.
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md]: You can just say Visual Studio 2022 here. No need to specify a version, as it's included in all versions (independent from MAUI).

File: docs/android/emulator/device-manager.md [hunk 19]
```diff
@@ -0,0 +1,426 @@
...
+  - **Android SDK Tools 5.0** or later
+  - **Android SDK Platform-Tools 31.0.3** or later
+  - **Android SDK Build-Tools 30.0.2** or later
+  - **Android Emulator 30.8.4** or later
+
+  These packages should be displayed with **Installed** status as seen in the following screenshot:
+
+  :::image type="content" source="media/device-manager/win/02-sdk-tools.png" alt-text="Installing Android SDK Tools.":::
+
+When you install the **Mobile development with .NET** workload in Visual Studio, and select the optional **.NET MAUI (Preview)** workload, everything is installed for you. For more information on setting up .NET MAUI with Visual Studio, see [Installation](../../get-started/installation.md).
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md]: I think you can just say that "When you install blah workload..., everything is installed for you". - no need to mention the optional workload. It's checked by default and this will save having to return to this doc to edit it out at GA.

File: docs/android/emulator/device-manager.md [hunk 20]
```diff
@@ -0,0 +1,426 @@
...
+
+## Open the device manager
+
+Open the Android Device Manager in Visual Studio from the **Tools** menu by pressing **Tools > Android > Android Device Manager**:
+
+:::image type="content" source="media/device-manager/win/03-tools-menu.png" alt-text="Launching the Device manager from the Tools menu.":::
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md:52]: This should be a default (light) theme IDE screenshot, right?

File: docs/android/emulator/device-manager.md [hunk 21]
```diff
@@ -0,0 +1,426 @@
...
+
+To create a new device, press the **New** button:
+
+:::image type="content" source="media/device-manager/win/08-new-button.png" alt-text="New button for creating a new device.":::
+
+The **New Device** window is displayed. To configure the device, follow these steps:
+
+01. Give the device a new name. In the following example, the new device is named _Pixel 2 - API 28_.
+
+    :::image type="content" source="media/device-manager/win/09-device-props-sml.png" alt-text="New Device screen of the Device Manager." lightbox="media/device-manager/win/09-device-props.png":::
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md:70]: It might be better to go with a Processor field of x86_64. Google Play Store has required x86_64 apps for a while now.

File: docs/android/emulator/troubleshooting.md [hunk 22]
```diff
@@ -0,0 +1,463 @@
...
+## Missing Google Play Services
+
+If the emulated Android device doesn't have Google Play Services or Google Play Store installed, you probably created a virtual device that excluded these packages. When you [create a virtual device](device-manager.md), be sure to select one or both of the following options:
+
+- **Google APIs**&mdash;includes Google Play Services in the virtual device.
+- **Google Play Store**&ndash;includes Google Play Store in the virtual device.
+
+For example, this virtual device will include Google Play Services and Google Play Store:
+
+:::image type="content" source="media/troubleshooting/win/00-add-gps-w158.png" alt-text="Android Device Manager with Google Play Services and Play Store enabled for .NET MAUI.":::
```
Comment (davidbritch) [inline docs/android/emulator/troubleshooting.md:58]: Use x86_64 processor.

File: docs/android/emulator/troubleshooting.md [hunk 23]
```diff
@@ -0,0 +1,463 @@
...
+
+### Hardware acceleration isn't enabled
+
+When you start a virtual device, and you don't have hardware acceleration enabled, the Device Manager displays an error dialog similar to the following image:
+
+:::image type="content" source="media/troubleshooting/win/01-dev-mgr-warning-w158.png" alt-text="Android device manager warning about Hyper-V not enabled on .NET MAUI.":::
+
+To fix this error, follow the troubleshooting steps in the [Hardware acceleration issues](#hardware-acceleration-issues) section.
+
+### Acceleration is enabled but the emulator runs too slowly
```
Comment (davidbritch) [inline docs/android/emulator/troubleshooting.md]: This should be x86_64.

File: docs/android/emulator/troubleshooting.md [hunk 24]
```diff
@@ -0,0 +1,463 @@
...
+
+#### Disabling Device Guard
+
+Device Guard and Credential Guard can prevent Hyper-V from being disabled on Windows machines. This situation is often a problem for domain-joined machines that are configured and controlled by an owning organization. On Windows 10, use the following steps to see if **Device Guard** is running:
+
+01. Enter **System info** in the Windows search box and select **System Information** in the search results.
+
+01. In the **System Summary**, look to see if **Device Guard Virtualization based security** is present and is in the **Running** state:
+
+   :::image type="content" source="media/troubleshooting/win/04-device-guard-sml.png" alt-text="Check if Device Guard is running for .NET MAUI." lightbox="media/troubleshooting/win/04-device-guard.png":::
```
Comment (davidbritch) [inline docs/android/emulator/troubleshooting.md]: Needs indenting further?

File: docs/android/emulator/troubleshooting.md [hunk 25]
```diff
@@ -0,0 +1,463 @@
...
+
+If Device Guard is enabled, use the following steps to disable it:
+
+01. Ensure that **Hyper-V** is disabled (under **Turn Windows Features on or off**) as described in the previous section.
+
+01. In the Windows Search Box, enter **gpedit.msc** and select the **Edit group policy** search result. These steps launch the **Local Group Policy Editor**.
+
+01. In the **Local Group Policy Editor**, navigate to **Computer Configuration > Administrative Templates > System > Device Guard**:
+
+   :::image type="content" source="media/troubleshooting/win/05-group-policy-editor-sml.png" alt-text="Check if Device Guard is enabled in the Local Group Policy editor for .NET MAUI." lightbox="media/troubleshooting/win/05-group-policy-editor.png":::
```
Comment (davidbritch) [inline docs/android/emulator/troubleshooting.md]: Needs indenting further?

**Review by davidbritch — CHANGES_REQUESTED** [review]
Two main issues:

1. There are a couple of dark IDE screenshots. These should be light, right?
2. Any text/screenshots that mentions x86 should really be x86_64. 

+ some tiny issues.

File: docs/TOC.yml (hunk 9 above)
Comment (adegeo) [reply docs/TOC.yml]: Ok, I'll remove them. Just for a demonstration of the behavior, they're only expanded when:
1. You're navigated to a page within Android
2. You manually expanded Android when you're outside of android
docs

File: docs/android/emulator/debug-on-emulator.md (hunk 14 above)
Comment (adegeo) [reply docs/android/emulator/debug-on-emulator.md:19]: I got this last year too when I did some WPF/WinForms articles. Testing on a fresh VM back then, the default IDE theme was dark.

File: docs/android/emulator/hardware-acceleration.md (hunk 15 above)
Comment (adegeo) [reply docs/android/emulator/hardware-acceleration.md]: It's confusing for sure. The original article states what is written here: x86 is the best image for speed. The guidance from Xamarin articles seem to say stay away from x64. Are they just old and super out of date?

File: docs/android/emulator/index.md (hunk 16 above)
Comment (adegeo) [reply docs/android/emulator/index.md]: I think it's because of the context. The article is speaking about configuring the emulator so you can start coding and working. But debugging is something you do after you're already off and running.

I'm not sure if we're supposed to even have these "link farm" articles around anymore. Theoretically, if the TOC title and search engine results can't get someone where they're supposed to be, we're doing it wrong 😁

File: docs/android/emulator/hardware-acceleration.md (hunk 15 above)
Comment (davidbritch) [reply docs/android/emulator/hardware-acceleration.md]: Yes. Any content under /ios or /android was handed to the product team back in 2018 and hence isn't up to date.

The x86 image might be faster (I honestly don't know) but the requirement to submit 64 bit versions of apps to the Play store has been in place since 2019.

File: docs/android/emulator/index.md (hunk 16 above)
Comment (davidbritch) [reply docs/android/emulator/index.md]: So to be honest I wouldn't have migrated the index.md article. For me, it adds no value. The Xamarin docs are full of such articles (a hangover from before the docs were on docs.microsoft.com), and in 95% of cases I've been deleting them as I migrate content. I just assumed that as you'd migrated it you saw some value in it.

**Comment by davidbritch** [comment]
@adegeo It's probably worth including the include file in each doc, that every other doc in the repo includes.

File: docs/android/emulator/device-manager.md [hunk 26]
```diff
@@ -0,0 +1,426 @@
...
+
+## Requirements
+
+To use the Android Device Manager, you'll need the following items:
+
+- Visual Studio 2022: Community, Professional, and Enterprise editions are supported.
+
+- The **Android SDK API Level 30** or later. Be sure to install the Android SDK at its default location if it isn't already installed: _C:\\Program Files (x86)\\Android\\android-sdk_.
+
+- The following packages must be installed in the:
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md]: in the?

File: docs/android/emulator/device-manager.md [hunk 27]
```diff
@@ -0,0 +1,426 @@
...
+
+### Create a new device
+
+To create a new device, press the **New** button:
+
+:::image type="content" source="media/device-manager/win/08-new-button.png" alt-text="New button for creating a new device.":::
+
+The **New Device** window is displayed. To configure the device, follow these steps:
+
+01. Give the device a new name. In the following example, the new device is named _Pixel 2 - API 28_.
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md]: It's API 31 in the screenshot :)

File: docs/android/emulator/device-manager.md [hunk 28]
```diff
@@ -0,0 +1,426 @@
...
+
+    :::image type="content" source="media/device-manager/win/09-device-props-sml.png" alt-text="New Device screen of the Device Manager." lightbox="media/device-manager/win/09-device-props.png":::
+
+01. Select a physical device to emulate by selecting a device in the **Base Device** box.
+01. Select a processor type for this virtual device with the **Processor** box.
+
+    Selecting **x86** will provide the best performance because it enables the emulator to take advantage of [hardware acceleration](hardware-acceleration.md). The **x86_64** option will also make use of hardware acceleration, but it runs slightly slower than **x86** (**x86_64** is normally used for testing 64-bit apps).
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md]: I'd reword this paragraph to drop the mention of x86 running faster (no idea if it's still true), and just say that x86_64 makes use of h/w accel.

File: docs/android/emulator/device-manager.md [hunk 29]
```diff
@@ -0,0 +1,426 @@
...
+
+### Additional options
+
+Additional options for working with devices are available from the **Additional Options** (&hellip;) pull-down menu:
+
+:::image type="content" source="media/device-manager/win/24-overflow-menu.png" alt-text="Location of additional options menu.":::
+
+The additional options menu contains the following items:
+
+- **Duplicate and Edit** &ndash; Duplicates the currently-selected device and opens it in the **New Device** screen with a new name that's similar to the existing device. For example, selecting **Pixel 2 - API 28** and pressing **Duplicate and Edit** appends a counter to the name: **Pixel 2 - API 28 (1)**.
```
Comment (davidbritch) [inline docs/android/emulator/device-manager.md]: Mention API 31 instead of 28 (seeing 31 was what was created earlier)?

File: docs/android/emulator/hardware-acceleration.md [hunk 30]
```diff
@@ -0,0 +1,227 @@
...
+ms.topic: how-to
+---
+
+# How to enable hardware acceleration with Android emulators (Hyper-V & HAXM)
+
+This article explains how to use your computer's hardware acceleration features to maximize Android emulator performance.
+
+[!INCLUDE [docs under construction](~/includes/preview-note.md)]
+
+With Visual Studio, you can easily test and debug your .NET MAUI app for Android in situations where an Android device isn't available. However, if hardware acceleration isn't available or enabled, the emulator will run too slow. You can drastically improve the performance of the emulator by enabling hardware acceleration and using special x86 virtual device images.
```
Comment (davidbritch) [inline docs/android/emulator/hardware-acceleration.md]: x86_64

Maybe drop "special"? Not sure what's special about them.

File: docs/android/emulator/hardware-acceleration.md [hunk 31]
```diff
@@ -0,0 +1,227 @@
...
+
+### Installing HAXM
+
+HAXM install packages for Windows are available from the [Intel Hardware Accelerated Execution Manager](https://github.com/intel/haxm/releases) GitHub releases page. Use the following steps to download and install HAXM:
+
+01. From the Intel website, download the latest [HAXM virtualization engine](https://github.com/intel/haxm/releases) installer for Windows. The advantage of downloading the HAXM installer directly from the Intel website is that you can be assured of using the latest version.
+
+01. Run **intelhaxm-android.exe** to start the HAXM installer. Accept the default values in the installer dialogs.
+
+When you [create a virtual device](device-manager.md), be sure to select an **x86**-based system image. If you use an ARM-based system image, the virtual device will not be accelerated and will run slowly.
```
Comment (davidbritch) [inline docs/android/emulator/hardware-acceleration.md]: x86_64

File: docs/android/emulator/hardware-acceleration.md [hunk 32]
```diff
@@ -0,0 +1,227 @@
...
+
+01. **Intel's Hardware Accelerated Execution Manager (HAXM)**.
+
+    HAXM is a virtualization engine for computers running Intel CPUs.
+
+For the best experience on Windows, it's recommended you use WHPX to accelerate the Android emulator. If WHPX isn't available on your computer, then HAXM can be used. The Android emulator automatically uses hardware acceleration if the following criteria are met:
+
+- Hardware acceleration is available and enabled on your development computer.
+
+- The emulator is running a system image created for an **x86**-based virtual device.
```
Comment (davidbritch) [inline docs/android/emulator/hardware-acceleration.md]: x86_64

File: docs/android/emulator/hardware-acceleration.md [hunk 33]
```diff
@@ -0,0 +1,227 @@
...
+01. Enter **windows features** in the Windows search box and select **Turn Windows features on or off** in the search results. In the **Windows Features** dialog, enable both **Hyper-V** and **Windows Hypervisor Platform**:
+
+    :::image type="content" source="media/hardware-acceleration/win/windows-features.png" alt-text="Enabling Hyper-V and Windows Hypervisor Platform for .NET MAUI.":::
+
+    After making these changes, reboot your computer.
+
+> [!IMPORTANT]
+> On Windows 10 October 2018 Update (RS5) and higher, you only need to enable Hyper-V, as it will use Windows Hypervisor Platform (WHPX) automatically.
+
+01. Make sure that the virtual device you [created in the Android Device Manager](device-manager.md) is an **x86**-based system image. If you use an Arm-based system image, the virtual device won't be accelerated and will run slowly.
```
Comment (davidbritch) [inline docs/android/emulator/hardware-acceleration.md]: x86_64

**Review by davidbritch — CHANGES_REQUESTED** [review]
Some last minor changes.

**Review by davidbritch — APPROVED** [review]
Thanks @adegeo
